shared/src/main/scala/mlscript/NuTypeDefs.scala

```diff
@@ -838,15 +1193,34 @@ class NuTypeDefs extends ConstraintSolver { self: Typer =>
             case _ => die
           }
         case td: NuTypeDef =>
-          typeSignatureOf(td, level, tparams, typedParams)
+          typeSignatureOf(td, level, tparams, typedParams, TopType, inheritedTags)
       }
     
     override def toString: String =
       s"${decl.name} ~> ${if (isComputing) "<computing>" else result.fold("<uncomputed>")(_.toString)}"
     
   }
   
-  
+  def memberUn(l: Ls[NuMember], r: Ls[NuMember])(implicit raise: Raise): Ls[NuMember] = {
+    val nms = Set.from(l.map(_.name) ++ r.map(_.name)).toList
+    nms.map {n => 
+      (l.find(x => x.name == n), r.find(x => x.name == n)) match {
+        case(S(a: TypedNuFun), S(b: TypedNuFun)) 
+          if a.level == b.level 
+            && a.fd.isLetRec == b.fd.isLetRec 
+            && a.fd.nme == b.fd.nme
+            && a.fd.tparams == b.fd.tparams
+            // todo: check fd.rhs
+          =>
+            TypedNuFun(a.level, a.fd, a.bodyType & b.bodyType)
+        case(S(a), N) => a
+        case(N, S(b)) => b
+        case _ => 
+          err(msg"invalid $n", N)
+          ???
```

Comment:
What does this error mean? Can it ever happen? If not, document why and use the `lastWords` function instead of `???`.